core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -1285,10 +1331,26 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord]) extends PSBTMap {
       case _: P2PKScriptPubKey =>
         collectSigs(required = 1, sigs => P2PKScriptSignature(sigs.head._1))
       case multiSig: MultiSignatureScriptPubKey =>
-        collectSigs(required = multiSig.requiredSigs,
-                    sigs => MultiSignatureScriptSignature(sigs.map(_._1)))
-      case EmptyScriptPubKey | _: NonStandardScriptPubKey |
-          _: UnassignedWitnessScriptPubKey | _: WitnessCommitment =>
+        def generateScriptSig(
+            sigs: Seq[(ECDigitalSignature, ECPublicKey)]): MultiSignatureScriptSignature = {
+          val sortedSigs = sigs
+            .map {
+              case (sig, pubKey) =>
+                (sig, pubKey, multiSig.publicKeys.indexOf(pubKey))
+            }
+            .sortBy(_._3)
+            .map(_._1)
+
+          MultiSignatureScriptSignature(sortedSigs)
+        }
+
+        collectSigs(required = multiSig.requiredSigs, generateScriptSig)
+      case EmptyScriptPubKey =>
+        // This script pushes an OP_TRUE onto the stack, causing a successful spend
+        val scriptSig = NonStandardScriptSignature("0151")
```

Comment:
I think this `NonstandardScriptSignature("0151")` is used else where, might be worthwhile to just refactor it to be some place with a nice name.